qualang_tools/addons/calibration/calibrations.py

```diff
@@ -0,0 +1,1208 @@
+                        # Second pi/2 pulse
+                        play(self.qubit_op, self.qubit_elmt)
+                        # Align, measure and save data
+                        self._align_measure_save(I, Q, I_st, Q_st, calib)
+                    save(n, n_st)
+
+                with stream_processing():
+                    I_st.buffer(len(var1_scan)).average().save("I")
+                    Q_st.buffer(len(var1_scan)).average().save("Q")
+                    n_st.save("iteration")
+
+            if self.simulation:
+                return prog_ramsey_1D
+            job = self.qm.execute(prog_ramsey_1D)
+
+            self._get_IQ_data_and_plot(job, calib, plot, scan_dimension)
+
+        elif scan_dimension == 2:
+            with program() as prog_ramsey_2D:
+
+                # Get the QUA variables for the 2D scan + averaging
+                (
+                    n,
+                    var1,
+                    var1_scan,
+                    var1_name,
+                    var2,
+                    var2_scan,
+                    var2_name,
+                ) = self._get_qua_variables(calib)
+
+                I = declare(fixed)
+                Q = declare(fixed)
+                I_st = declare_stream()
+                Q_st = declare_stream()
+                n_st = declare_stream()
+
+                with for_(n, 0, n < self.scan_var[calib]["averaging"], n + 1):
+                    with for_(*from_array(var2, var2_scan)):
+                        with for_(*from_array(var1, var1_scan)):
+                            # Update the resonator frequency
+                            if var1_name == "frequency":
+                                update_frequency(self.qubit_elmt, var1)
+                            elif var2_name == "frequency":
+                                update_frequency(self.qubit_elmt, var2)
+                            # First pi/2 pulse
+                            play(self.qubit_op, self.qubit_elmt)
+                            # Adjust the idle time
+                            if var1_name == "duration":
+                                wait(var1, self.qubit_elmt)
+                            elif var2_name == "duration":
+                                wait(var2, self.qubit_elmt)
+                            elif self.user_specifics[calib]["idle_time"] is not None:
+                                wait(
+                                    self.user_specifics[calib]["idle_time"],
+                                    self.qubit_elmt,
+                                )
+                            # Second pi/2 pulse
+                            play(self.qubit_op, self.qubit_elmt)
+
+                            # Align, measure and save data
+                            self._align_measure_save(I, Q, I_st, Q_st, calib)
+                    save(n, n_st)
+
+                with stream_processing():
+                    I_st.buffer(len(var2_scan), len(var1_scan)).average().save("I")
```

Comment:
maybe we can change it to `I_st.buffer(len(var1_scan)).buffer(len(var2_scan))` this is more intutitive for me

Reply:
Alright, the syntax is `I_st.buffer(1st_inner_loop).buffer(2nd_inner_loop).buffer(3rd_inner_loop)...` right?